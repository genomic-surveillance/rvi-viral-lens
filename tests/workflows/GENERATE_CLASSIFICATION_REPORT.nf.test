nextflow_workflow {

    name "Test Workflow GENERATE_CLASSIFICATION_REPORT"
    script "workflows/GENERATE_CLASSIFICATION_REPORT.nf"
    workflow "GENERATE_CLASSIFICATION_REPORT"

    test("Should run without failures") {

        when {
            params {
                meta = [
                    sample_id:"test",
                    taxid:2222222,
                    taxid_name:"test_taxid_name",
                    virus_species:"SARSCov2",
                    virus_subtype:"BA.2",
                    flu_segment:"",
                    percentage_genome_coverage:90,
                    total_mapped_reads:200
                ]
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.of(tuple(params.meta))
                """
            }
        }

        then {
            assert workflow.success
        }

    }

}