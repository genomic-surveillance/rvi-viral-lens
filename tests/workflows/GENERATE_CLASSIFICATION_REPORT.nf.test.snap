{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "classification_report.csv:md5,1e51062e8c631db04ff690b79444f6d1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-04-23T15:08:25.677299"
    }
}